### PR TITLE
port crm_claim_rma to 8.0

### PR DESCRIPTION
**Pull request by osvalr**
This PR ports crm_claim_rma to 8.0 and OCA Guidelines.

**`UPDATE`**
The following video shows with its basic functioning.
[Spanish Audio](http://www.youtube.com/watch?v=Eu5cj0HlCh4)

### CODE REVIEW

**Comment by osvalr** [comment]
@yaniaular @pedrobaeza can you review this PR please?

File: crm_claim_rma/models/account_invoice.py [hunk 1]
```diff
@@ -22,21 +23,23 @@
...
 
 
 class AccountInvoice(models.Model):
+    """
+    Account Invoice
+    """
     _inherit = "account.invoice"
 
-    claim_id = fields.Many2one('crm.claim', string='Claim')
+    claim_id = fields.Many2one('crm.claim', string='Claim', translate=True)
```
Comment (pedrobaeza) [inline crm_claim_rma/models/account_invoice.py]: A many2one is not translatable

File: crm_claim_rma/models/account_invoice.py [hunk 2]
```diff
@@ -51,6 +54,7 @@ def _refund_cleanup_lines(self, lines):
             if not claim_line.refund_line_id:
                 # For each lines replace quantity and add claim_line_id
                 inv_line = claim_line.invoice_line_id
+
```
Comment (pedrobaeza) [inline crm_claim_rma/models/account_invoice.py]: Don't introduce unneeded line feeds

File: crm_claim_rma/models/account_invoice_line.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
```
Comment (pedrobaeza) [inline crm_claim_rma/models/account_invoice_line.py:1]: You don't need to split this submodel in another file. The rule only applies to main models, but not by this kind of submodels that are always attached to its parent.

File: crm_claim_rma/models/claim_line.py [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+                               readonly=True,
+                               help="If warranty has expired")
+
+    @api.model
+    def get_warranty_return_partner(self):
+        return self.env['product.supplierinfo'].get_warranty_return_partner()
+
+    warranty_type = fields.Selection(get_warranty_return_partner,
+                                     readonly=True,
+                                     help="Who is in charge of the warranty "
```
Comment (pedrobaeza) [inline crm_claim_rma/models/claim_line.py]: To use less lines of code, you can put first argument in the next line and then start with only one indentation more

File: crm_claim_rma/models/claim_line.py [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+        vals = vals or {}
+
+        if ('number' not in vals) or (vals.get('number', False) == '/'):
+            vals['number'] = self._get_sequence_number()
+
+        res = super(ClaimLine, self).create(vals)
+        return res
+
+    @api.multi
+    def name_get(self):
```
Comment (pedrobaeza) [inline crm_claim_rma/models/claim_line.py]: Better to use a compute field called `display_name` and change _rec_name to it.

File: crm_claim_rma/models/crm_claim.py [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+            'crm.claim'))
+
+    claim_line_ids = fields.One2many('claim.line', 'claim_id',
+                                     string='Return lines')
+    planned_revenue = fields.Float('Expected revenue')
+    planned_cost = fields.Float('Expected cost')
+    real_revenue = fields.Float()
+    real_cost = fields.Float()
+    invoice_ids = fields.One2many(
+        'account.invoice',
```
Comment (pedrobaeza) [inline crm_claim_rma/models/crm_claim.py]: You don't need to put one argument per line. You can join several arguments in the same line if they fits in 80 cols.

File: crm_claim_rma/models/account_invoice_line.py (hunk 3 above)
Comment (osvalr) [inline crm_claim_rma/models/account_invoice_line.py:1]: Hi @pedrobaeza, 

As [File Naming](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#file-naming) and [Differences With Odoo Guidelines](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#differences-with-odoo-guidelines) sections in Contributing guidelines shows, any inherited model is a separated file.

Comment (pedrobaeza) [inline crm_claim_rma/models/account_invoice_line.py:1]: The key in the guidelines is the word "main_model", that tells about what I'm saying. I remember to have more explicit instructions about it, but now I can't find it. We can make an amendment in the guidelines to say it more explicitly

File: crm_claim_rma/models/claim_line.py (hunk 5 above)
Comment (osvalr) [inline crm_claim_rma/models/claim_line.py]: Thanks, I didn't about it.

**Comment by yaniaular** [comment]
:+1: IMHO LGTM

File: crm_claim_rma/models/account_invoice_line.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+#
+##############################################################################
+from openerp import api, models
+
+
+class AccountInvoiceLine(models.Model):
+
+    """
+    Account Invoice Line
+    """
```
Comment (nhomar) [inline crm_claim_rma/models/account_invoice_line.py]: Unnecesary doc, just doc classes if you will put information with meaning, if you will only duplicate the class name itself, it is better to not put any doc here.

File: crm_claim_rma/tests/test_picking_creation.py [hunk 8]
```diff
@@ -125,7 +126,7 @@ def test_02_new_product_return(self):
         warehouse_rec = \
             warehouse_obj.search([('company_id',
                                    '=', company.id)])[0]
-        wizard = self.wizard_make_picking.with_context({
+        wizard = self.wizardmakepicking.with_context({
```
Comment (nhomar) [inline crm_claim_rma/tests/test_picking_creation.py]: why not _ _? this name is confusing.

**Comment by osvalr** [comment]
@moylop260 can you help me with this? Travis-CI failed and nothing changed in its configuration from my part. Compare travis jobs for commits: [17ba63e](https://travis-ci.org/OCA/rma/jobs/89303810#L204) vs [dc67334](https://travis-ci.org/OCA/rma/jobs/89492131#L208), it seems that `/home/travis/crm,/home/travis/stock-logistics-workflow` from addons path is missing in the 2nd one.

**Comment by moylop260** [comment]
@osvalr 
see next issue: https://github.com/OCA/maintainer-quality-tools/issues/291

**Comment by nhomar** [comment]
:+1:

**Comment by yaniaular** [comment]
:+1:

**Comment by rvalyi** [comment]
cannot take time to review now, but I just wanted to say kudos for the amount of great work. I know how this module started; not that nice. But now it looks so much better...

**Comment by aristobulo-eezee** [comment]
A bit too late but -1

You took Javier Carrasco out of contributors list, I know you are in a rush but giving yourself upvotes that quick is not cool.

**Comment by pedrobaeza** [comment]
Please make a PR including again him as contributor. The PR has been here since 15 oct, and the last commit was 11 days ago, so it hasn't been so quick...

**Comment by nhomar** [comment]
@aristobulo-eezee you are right, but it was not intentionally it was a mistake (sorry for that), @osvalr can you correct the authorship please in another PR?.

Thanks.

**Comment by osvalr** [comment]
@aristobulo-eezee @nhomar #66 adds him as a contributor

**Pull request merged**